source/opt.d

```diff
@@ -1,31 +1,64 @@
 import std.getopt;
+import std.variant;
+import std.file;
 
-void getoptions(string[] args)
+
+struct Options {
+
+		ulong countEpoch;
+		float mutation;
+		string input;
+		float probability;
+
+		this(string inp, float mutate, ulong  count, float prob)
+		{
+			this.input = inp;
+			this.mutation = mutate;
+			this.countEpoch = count;
+			this.probability = prob;
+		}
+}
+
+class NoInputParams : Exception
+{
+    this(string msg, string file = __FILE__, size_t line = __LINE__) {
+        super(msg, file, line);
+    }
+}
+
+Options getOptions(string[] args)
 in
 {
-    assert(args.length > 1, "Inputed empty paramters");
     assert(args.length < 5, "Opps too many argmunt inputed");
 }
 body
 {
-    string data;
-    int count_epoch;
-    int mutation_number;
-    float probability = 0.0;
+		
+    string input;
+    ulong countEpoch;
+    float mutationNumber;
+    float probability = 1.0;
     auto helpInformation = getopt(
         args,
-        "epoch|e","Counting epoch, number(int).", &count_epoch,
-        "mutation|m","Mutation, how large mutation should be(int).", &mutation_number,
+        "epoch|e","Counting epoch, number(int).", &countEpoch,
+        "mutation|m","Mutation, how large mutation should be(int).", &mutationNumber,
         "probability|p","Number(float) representing probability.", &probability,
-        "input|i","Input file with data.", &data
+        "input|i","Input file with data.", &input
     );
-    import std.stdio;
-    data.writeln;
+
+		if((getcwd() ~ "/" ~ input).exists == false)
+		{
+			throw new NoInputParams("Inputed image file " ~ input ~ " not found");
+		}
+
+		auto data = Options(input, mutationNumber, countEpoch, probability);
```

Comment:
Technicky vzato jsou to jedničky a nuly a tudíž data, ale v kontextu tohoto programu jsou data např. vstupní obrázek nebo genomy.